pkg/miscellaneous/miscellaneous.go

```diff
@@ -535,41 +535,62 @@ func ReadConfigFileAsMap(path string) (map[string]interface{}, error) {
 	return c, nil
 }
 
-// ParsePeerURL forms a PeerURL, given podName by parsing the initial-advertise-peer-urls
-func ParsePeerURL(initialAdvertisePeerURLs, podName string) (string, error) {
-	tokens := strings.Split(initialAdvertisePeerURLs, "@")
-	if len(tokens) < 4 {
-		return "", fmt.Errorf("invalid peer URL : %s", initialAdvertisePeerURLs)
-	}
-	domaiName := fmt.Sprintf("%s.%s.%s", tokens[1], tokens[2], "svc")
-	return fmt.Sprintf("%s://%s.%s:%s", tokens[0], podName, domaiName, tokens[3]), nil
-}
-
-// IsPeerURLTLSEnabled checks whether the peer address is TLS enabled or not.
-func IsPeerURLTLSEnabled() (bool, error) {
-	podName, err := GetEnvVarOrError("POD_NAME")
+// GetAdvertiseURLs returns the advertise URLs of desired type for the etcd member.
+func GetAdvertiseURLs(advURLType string, configFile string) (string, error) {
+	memberName, err := GetEnvVarOrError("POD_NAME")
 	if err != nil {
-		return false, err
+		return "", err
 	}
-
-	configFile := GetConfigFilePath()
-
 	config, err := ReadConfigFileAsMap(configFile)
 	if err != nil {
-		return false, err
+		return "", err
+	}
+	advertiseURLs := config[advURLType]
+	if advertiseURLs == nil {
+		return "", fmt.Errorf("%s must be set in etcd config", advURLType)
+	}
+
+	podUrlsMap, ok := advertiseURLs.(map[interface{}]interface{})
+	if !ok {
+		return "", fmt.Errorf("%s is not in the expected format", advURLType)
+	}
+	resultMap := make(map[string][]string)
+	for pod, urls := range podUrlsMap {
+		podName, ok := pod.(string)
+		if !ok {
+			return "", fmt.Errorf("pod name is not a string")
+		}
+		urlsList, ok := urls.([]interface{})
+		if !ok {
+			return "", fmt.Errorf("urls is not a list")
+		}
+		for _, url := range urlsList {
+			urlStr, ok := url.(string)
+			if !ok {
+				return "", fmt.Errorf("url is not a string")
+			}
+			resultMap[podName] = append(resultMap[podName], urlStr)
+		}
+	}
+	podUrls, ok := resultMap[memberName]
+	if !ok {
```

Comment:
I have previously used two functions like you suggested, but reverted thinking it would be redundant but I agree it is much more readable to have separate functions. Hence, I have now followed your suggestion.